manager/state/raft/raft.go

```diff
@@ -224,15 +221,32 @@ func NewNode(ctx context.Context, opts NewNodeOptions) *Node {
 	return n
 }
 
+// WithContext returns context which is cancelled when parent context cancelled
+// or node is stopped.
+func (n *Node) WithContext(ctx context.Context) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(ctx)
+
+	go func() {
+		select {
+		case <-ctx.Done():
+		case <-n.stopped:
+			cancel()
+		}
+	}()
```

Comment:
To be honest, I'm not really excited about having a goroutine for every context we derive in this way. There's already a lot of goroutines that show up in a dump with SwarmKit running, and it gets kind of confusing to deal with so many. Most of them aren't really doing anything, but just sitting and waiting in case some specific thing happens. This goroutine would fall into that category.

But I can't think of a better way to do this, and I don't want to hold up the cleanup.